com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/MemoryMetricsSource.java

```diff
@@ -0,0 +1,535 @@
+/* Generated by Streams Studio: October 18, 2016 at 3:12:58 PM GMT+2 */
+package com.ibm.streamsx.metrics;
+
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.management.JMX;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.apache.log4j.Logger;
+
+import com.ibm.streams.management.Metric;
+import com.ibm.streams.management.ObjectNameBuilder;
+import com.ibm.streams.management.instance.InstanceMXBean;
+import com.ibm.streams.management.job.JobMXBean;
+import com.ibm.streams.management.job.PeMXBean;
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.OutputTuple;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingOutput;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.OutputPortSet;
+import com.ibm.streams.operator.model.OutputPortSet.WindowPunctuationOutputMode;
+import com.ibm.streams.operator.model.OutputPorts;
+import com.ibm.streams.operator.model.Parameter;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+import com.ibm.streamsx.metrics.internal.OperatorConfiguration;
+import com.ibm.streamsx.metrics.internal.TupleContainer;
+
+/**
+ * A source operator that does not receive any input streams and produces new tuples. 
+ * The method <code>produceTuples</code> is called to begin submitting tuples.
+ * <P>
+ * For a source operator, the following event methods from the Operator interface can be called:
+ * </p>
+ * <ul>
+ * <li><code>initialize()</code> to perform operator initialization</li>
+ * <li>allPortsReady() notification indicates the operator's ports are ready to process and submit tuples</li> 
+ * <li>shutdown() to shutdown the operator. A shutdown request may occur at any time, 
+ * such as a request to stop a PE or cancel a job. 
+ * Thus the shutdown() may occur while the operator is processing tuples, punctuation marks, 
+ * or even during port ready notification.</li>
+ * </ul>
+ * <p>With the exception of operator initialization, all the other events may occur concurrently with each other, 
+ * which lead to these methods being called concurrently by different threads.</p> 
+ */
+@PrimitiveOperator(
```

Comment:
This operator becomes obsolete with the latest (already merged) MetricsSource extension. The MetricsSource operator now supports also PE metrics.